src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -111,25 +111,25 @@ end
 ##### Tuple fill_halo! kernels
 #####
 
-@kernel function _fill_west_and_east_halo!(c::NTuple{N}, west_bc::NTuple{N}, east_bc::NTuple{N}, grid, args...) where N
+@kernel function _fill_west_and_east_halo!(c::Tuple, west_bc::Tuple, east_bc::Tuple, grid, args...) 
     j, k = @index(Global, NTuple)
-    @unroll for n in 1:N
+    @unroll for n in 1:length(c)
```

Comment:
Humm...

If that doesn't work you could try passing the length explicitly via `Val(N)` and extracting with `::Val{N}`